coldfront/core/project/management/commands/add_service_units_to_project.py

```diff
@@ -0,0 +1,115 @@
+                      f'{project.name}. This would increase {project.name} ' \
+                      f'SUs from {current_allocation} to {allocation}. ' \
+                      f'The reason for updating SUs for {project.name} ' \
+                      f'would be: {reason}.'
+
+            self.stdout.write(self.style.WARNING(message))
+
+        else:
+            # Set the value for the Project.
+            set_project_allocation_value(project, allocation)
+
+            # Create a transaction to record the change.
+            ProjectTransaction.objects.create(
+                project=project,
+                date_time=date_time,
+                allocation=allocation)
+
+            # Set the reason for the change in the newly-created historical object.
+            allocation_objects.allocation_attribute.refresh_from_db()
+            historical_allocation_attribute = \
+                allocation_objects.allocation_attribute.history.latest("id")
+            historical_allocation_attribute.history_change_reason = reason
+            historical_allocation_attribute.save()
+
+            # Do the same for each ProjectUser.
+            allocation_attribute_type = AllocationAttributeType.objects.get(
+                name="Service Units")
+            for project_user in project.projectuser_set.all():
+                user = project_user.user
+                # Attempt to set the value for the ProjectUser. The method returns whether
+                # it succeeded; it may not because not every ProjectUser has a
+                # corresponding AllocationUser (e.g., PIs). Only proceed with further steps
+                # if an update occurred.
+                allocation_updated = set_project_user_allocation_value(
+                    user, project, allocation)
+                if allocation_updated:
+                    # Create a transaction to record the change.
+                    ProjectUserTransaction.objects.create(
+                        project_user=project_user,
+                        date_time=date_time,
+                        allocation=allocation)
+                    # Set the reason for the change in the newly-created historical object.
+                    allocation_user = \
+                        allocation_objects.allocation.allocationuser_set.get(user=user)
+                    allocation_user_attribute = \
+                        allocation_user.allocationuserattribute_set.get(
+                            allocation_attribute_type=allocation_attribute_type,
+                            allocation=allocation_objects.allocation)
+                    historical_allocation_user_attribute = \
+                        allocation_user_attribute.history.latest("id")
+                    historical_allocation_user_attribute.history_change_reason = reason
+                    historical_allocation_user_attribute.save()
+
+            message = f"Successfully added {addition} SUs to {project.name}" \
```

Comment:
Note the change for the users as well: "Successfully added {addition} SUs to {project.name} **and its users**, updating SUs from {current_allocation} to {allocation}."

Let's write this message to the log as well.